tests/general/HTTPSBearTest.py

```diff
@@ -0,0 +1,138 @@
+import io
+from queue import Queue
+import requests
+import requests_mock
+import unittest
+import unittest.mock
+
+from bears.general.HTTPSBear import HTTPSBear
+from bears.general.URLBear import URLBear
+from coalib.testing.LocalBearTestHelper import LocalBearTestHelper
+from coalib.settings.Section import Section
+
+
+def custom_matcher(request):
+    """
+    Mock the status codes for every request by taking the last three characters
+    of the request URL.
+
+    For connection checking url, it always passes 200 (prerequisite checking).
+
+    For URLs with no status codes appended, a ``RequestException`` is raised.
+
+    To test for URLs that redirect to URLs with a much larger length, the
+    ``redirect_long_url`` is returned.
+
+    :param request: The ``request`` that the mocker recieves.
+    :return:        A mocked ``Response`` object.
+    """
+
+    # the connection check url needs to be explicitly
+    # set to 200
+    if request.url == URLBear.check_connection_url:
+        status_code = 200
+    else:
+        try:
+            status_code = int(request.path_url[-3:])
+        except ValueError:
+            raise requests.exceptions.RequestException
+    resp = requests.Response()
+    resp.raw = io.BytesIO()
+    resp.status_code = status_code
+    return resp
```

Comment:
Isn't this has the same as [`custom_matcher` in InvalidLinkBear](https://github.com/coala/coala-bears/blob/master/tests/general/InvalidLinkBearTest.py#L17-L56) ?

If so, you could import that instead.

Reply:
This is not the same custom_matcher, it has been modified as per the https bear's requirement